### PR TITLE
Documentation for 3.1.0

### PR DESCRIPTION
**Pull request by felixs88**
# Description
Documentation update for Release: 3.1.0

# GitHub Issues
List the GitHub issues impacted by this PR:

| GitHub Issue # |
| -------------- |
| |

# Checklist:

- [ ] I have performed a self-review of my own code to ensure there are no formatting, pep8, linting, or security issues
- [ ] I have performed Ansible Sanity test using --docker default
- [ ] I have verified that new and existing unit tests pass locally with my changes
- [ ] I have not allowed coverage numbers to degenerate
- [ ] I have maintained at least 90% code coverage
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Backward compatibility is not broken

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -4,6 +4,20 @@ Dellemc.Powerscale Change Logs
...
+
+Minor Changes
+-------------
+
+- Added support for running as root, along with other boolean flags such as allow_delete_readonly, allow_execute_always, and inheritable_path_acl.
+
+New Modules
+-----------
+
+- dellemc.powerscale.roles - Manage auth roles on a PowerScale Storage System.
```
Comment (Bhavneet-Sharma) [inline CHANGELOG.rst:19]: Support assist is missing

File: README.md [hunk 2]
```diff
@@ -68,52 +76,53 @@ This table provides information about the software prerequisites for the Ansible
 
 | **Ansible Modules** | **OneFS Version** | **Python version** | **Python SDK version** | **Ansible**              |
 |---------------------|-----------------------|--------------------|----------------------------|--------------------------|
-| v3.0.0 | 9.4.x <br> 9.5.x <br> 9.7.x | 3.9 <br> 3.10 <br> 3.11 | 0.3.0.1 | 2.14 <br> 2.15 <br> 2.16 |
+| vmain | 9.4.x <br> 9.5.x <br> 9.7.x <br> 9.8.x | 3.9 <br> 3.10 <br> 3.11 <br> 3.11 | 0.3.0.1 | 2.14 <br> 2.15 <br> 2.16 <br> 2.16 |
```
Comment (Bhavneet-Sharma) [inline README.md:79]:
```suggestion
| main | 9.4.x <br> 9.5.x <br> 9.7.x <br> 9.8.x | 3.9 <br> 3.10 <br> 3.11 <br> 3.11 | 0.3.0.1 | 2.14 <br> 2.15 <br> 2.16 <br> 2.16 |
```

File: changelogs/changelog.yaml [hunk 3]
```diff
@@ -269,3 +269,14 @@ releases:
       minor_changes:
         - Added support for OneFS 9.7 Key West release.
     release_date: '2024-03-29'
+  3.1.0:
+    changes:
+      minor_changes:
+        - Added support for running as root, along with other boolean flags such
+          as allow_delete_readonly, allow_execute_always, and inheritable_path_acl.
+        - Added support for OneFS 9.8 Lonestar release.
+    modules:
```
Comment (Bhavneet-Sharma) [inline changelogs/changelog.yaml:278]: Support assist module is missing

**Review by Bhavneet-Sharma — COMMENTED** [review]
Please do add the RST files as well

**Review by sachin-apa — DISMISSED** [review]

File: galaxy.yml [hunk 4]
```diff
@@ -14,7 +14,7 @@ name: powerscale
 
 # The version of the collection.
 # Must be compatible with semantic versioning
-version: 3.0.0
+version: main
```
Comment (Bhavneet-Sharma) [inline galaxy.yml:17]:
```suggestion
version: 3.1.0
```

File: CHANGELOG.rst (hunk 1 above)
Comment (felixs88) [reply CHANGELOG.rst:20]: Done

File: changelogs/changelog.yaml (hunk 3 above)
Comment (felixs88) [reply changelogs/changelog.yaml:278]: Done

File: docs/Release Notes.md [hunk 5]
```diff
@@ -66,16 +66,18 @@ The Ansible Modules for Dell PowerScale support the following features:
 - Get details and modify SMB global settings.
 - Get details and modify SNMP settings.
 - Import, modify, setting default, and delete a server certificate.
+- Get details, create, modify, delete of auth roles.
```
Comment (Bhavneet-Sharma) [inline docs/Release Notes.md:70]: Add details about the support assist

File: galaxy.yml (hunk 4 above)
Comment (felixs88) [reply galaxy.yml]: Done.

File: docs/Release Notes.md (hunk 5 above)
Comment (felixs88) [reply docs/Release Notes.md:71]: Done

File: docs/Release Notes.md [hunk 6]
```diff
@@ -66,16 +66,20 @@ The Ansible Modules for Dell PowerScale support the following features:
...
 ---------------------------
 This section describes the features of the Ansible Modules for Dell PowerScale for this release.
 
-The Ansible Modules for Dell PowerScale release 3.0.0 supports the following features:
+The Ansible Modules for Dell PowerScale release 3.1.0 supports the following features:
 
-- Added support for OneFS version 9.7.x (Key West version).
+- Added support for Get, create, modify, delete auth roles.
+- Added support for Get, and modify support assist settings.
+- Added support for running as root, allow_delete_readonly, allow_execute_always, inheritable_path_acl.
```
Comment (Bhavneet-Sharma) [inline docs/Release Notes.md:83]: Added support for OneFS 9.8 Lonestar release.

Comment (felixs88) [reply docs/Release Notes.md:84]: Done

**Review by Bhavneet-Sharma — APPROVED** [review]

File: CHANGELOG.rst [hunk 7]
```diff
@@ -4,6 +4,22 @@ Dellemc.Powerscale Change Logs
...
 
+
+v3.1.0
+======
+
+Minor Changes
+-------------
+
+- Added support for OneFS 9.8 Lonestar release.
+- Added support for running as root, along with other boolean flags such as allow_delete_readonly, allow_execute_always, and inheritable_path_acl.
```
Comment (trisha-dell) [inline CHANGELOG.rst:15]: This is for SMB share module, please mention that.

Comment (felixs88) [reply CHANGELOG.rst]: yes updated

**Review by Bhavneet-Sharma — DISMISSED** [review]

File: README.md [hunk 8]
```diff
@@ -68,52 +78,54 @@ This table provides information about the software prerequisites for the Ansible
...
+  * [NFS Default Settings Module](https://github.com/dell/ansible-powerscale/blob/main/docs/modules/nfs_default_settings.rst)
+  * [NFS Global Settings Module](https://github.com/dell/ansible-powerscale/blob/main/docs/modules/nfs_global_settings.rst)
+  * [NFS Zone Settings Module](https://github.com/dell/ansible-powerscale/blob/main/docs/modules/nfs_zone_settings.rst)
+  * [SyncIQ Global Settings Module](https://github.com/dell/ansible-powerscale/blob/main/docs/modules/synciq_global_settings.rst)
+  * [SyncIQ Certificate Module](https://github.com/dell/ansible-powerscale/blob/main/docs/modules/synciqcertificate.rst)
+  * [SMB Global Settings Module](https://github.com/dell/ansible-powerscale/blob/main/docs/modules/smb_global_settings.rst)
+  * [SNMP Settings Module](https://github.com/dell/ansible-powerscale/blob/main/docs/modules/snmp_settings.rst)
+  * [Server Certificate Module](https://github.com/dell/ansible-powerscale/blob/main/docs/modules/server_certificate.rst)
+  * [Roles](https://github.com/dell/ansible-powerscale/blob/main/docs/modules/roles.rst)
+  * [Roles](https://github.com/dell/ansible-powerscale/blob/main/docs/modules/support_assist.rst)
```
Comment (trisha-dell) [inline README.md:124]: Change the title to Support Assist.

File: docs/Release Notes.md [hunk 9]
```diff
@@ -66,16 +66,21 @@ The Ansible Modules for Dell PowerScale support the following features:
 - Get details and modify SMB global settings.
 - Get details and modify SNMP settings.
 - Import, modify, setting default, and delete a server certificate.
+- Get details, create, modify, delete of auth roles.
```
Comment (trisha-dell) [inline docs/Release Notes.md:69]: Get details, create, modify, and delete auth roles.

File: docs/Release Notes.md [hunk 10]
```diff
@@ -66,16 +66,21 @@ The Ansible Modules for Dell PowerScale support the following features:
...
+- Get details, and modify support assist settings.
```
Comment (trisha-dell) [inline docs/Release Notes.md:70]: Get details, accept terms, and modify support assist settings.

File: docs/Release Notes.md [hunk 11]
```diff
@@ -66,16 +66,21 @@ The Ansible Modules for Dell PowerScale support the following features:
...
 New Features and Enhancements
 ---------------------------
 This section describes the features of the Ansible Modules for Dell PowerScale for this release.
 
-The Ansible Modules for Dell PowerScale release 3.0.0 supports the following features:
+The Ansible Modules for Dell PowerScale release 3.1.0 supports the following features:
 
-- Added support for OneFS version 9.7.x (Key West version).
+- Added support for OneFS 9.8 Lonestar release.
+- Added support for Get, create, modify, delete auth roles.
```
Comment (trisha-dell) [inline docs/Release Notes.md:81]: Added support to get, create, modify, and delete auth roles.

File: docs/Release Notes.md [hunk 12]
```diff
@@ -66,16 +66,21 @@ The Ansible Modules for Dell PowerScale support the following features:
...
+- Added support for Get, and modify support assist settings.
```
Comment (trisha-dell) [inline docs/Release Notes.md:82]: Added support to get details, accept terms, and modify support assist settings.

File: README.md [hunk 13]
```diff
@@ -68,52 +78,54 @@ This table provides information about the software prerequisites for the Ansible
 
 | **Ansible Modules** | **OneFS Version** | **Python version** | **Python SDK version** | **Ansible**              |
 |---------------------|-----------------------|--------------------|----------------------------|--------------------------|
-| v3.0.0 | 9.4.x <br> 9.5.x <br> 9.7.x | 3.9 <br> 3.10 <br> 3.11 | 0.3.0.1 | 2.14 <br> 2.15 <br> 2.16 |
+| main | 9.4.x <br> 9.5.x <br> 9.7.x <br> 9.8.x | 3.9 <br> 3.10 <br> 3.11 <br> 3.11 | 0.3.0.1 | 2.14 <br> 2.15 <br> 2.16 <br> 2.16 |
```
Comment (trisha-dell) [inline README.md:81]: Python version 3.11 is repeated twice, add 3.12 instead same goes for ansible 2.16. Please fix these.

File: README.md [hunk 14]
```diff
@@ -58,7 +68,7 @@ The tasks can be executed by running simple playbooks written in yaml syntax.
 * [Maintanence](#maintanence)
 
 ## License
-Ansible collection for PowerScale is released and licensed under the GPL-3.0 license. See [LICENSE](https://github.com/dell/ansible-powerscale/blob/3.0.0/LICENSE) for the full terms. Ansible modules and modules utilities that are part of the Ansible collection for PowerScale are released and licensed under the Apache 2.0 license. See [MODULE-LICENSE](https://github.com/dell/ansible-powerscale/blob/3.0.0/MODULE-LICENSE) for the full terms.
+Ansible collection for PowerScale is released and licensed under the GPL-3.0 license. See [LICENSE](https://github.com/dell/ansible-powerscale/blob/main/LICENSE) for the full terms. Ansible modules and modules utilities that are part of the Ansible collection for PowerScale are released and licensed under the Apache 2.0 license. See [MODULE-LICENSE](https://github.com/dell/ansible-powerscale/blob/main/MODULE-LICENSE) for the full terms.
 
 ## Supported platforms
   * Dell PowerScale OneFS versions 9.3.x, 9.4.x and 9.5.x
```
Comment (trisha-dell) [inline README.md:74]: 9.4.x, 9.5.x, 9.7.x and 9.8.x

File: README.md (hunk 8 above)
Comment (felixs88) [reply README.md]: Yes updated

File: docs/Release Notes.md (hunk 9 above)
Comment (felixs88) [reply docs/Release Notes.md]: yes, updated

File: docs/Release Notes.md (hunk 10 above)
Comment (felixs88) [reply docs/Release Notes.md]: yes, updated

File: README.md (hunk 13 above)
Comment (felixs88) [reply README.md]: yes, updated

File: docs/Release Notes.md (hunk 11 above)
Comment (felixs88) [reply docs/Release Notes.md]: yes, updated

File: docs/Release Notes.md (hunk 12 above)
Comment (felixs88) [reply docs/Release Notes.md]: yes, updated

**Review by sachin-apa — DISMISSED** [review]
Doc frag changes and rst generation needs update

File: README.md [hunk 15]
```diff
@@ -58,62 +68,64 @@ The tasks can be executed by running simple playbooks written in yaml syntax.
...
-  * Dell PowerScale OneFS versions 9.3.x, 9.4.x and 9.5.x
+  * Dell PowerScale OneFS versions 9.3.x, 9.4.x, 9.5.x, and 9.8.x
 
 ## Prerequisites
 This table provides information about the software prerequisites for the Ansible Modules for Dell PowerScale.
 
 | **Ansible Modules** | **OneFS Version** | **Python version** | **Python SDK version** | **Ansible**              |
 |---------------------|-----------------------|--------------------|----------------------------|--------------------------|
-| v3.0.0 | 9.4.x <br> 9.5.x <br> 9.7.x | 3.9 <br> 3.10 <br> 3.11 | 0.3.0.1 | 2.14 <br> 2.15 <br> 2.16 |
+| main | 9.4.x <br> 9.5.x <br> 9.7.x <br> 9.8.x | 3.9 <br> 3.10 <br> 3.11 <br> 3.12 | 0.3.0.1 | 2.14 <br> 2.15 <br> 2.16 <br> 2.17 |
```
Comment (Bhavneet-Sharma) [inline README.md:81]:
```suggestion
| v3.1.0| 9.4.x <br> 9.5.x <br> 9.7.x <br> 9.8.x | 3.9 <br> 3.10 <br> 3.11 <br> 3.12 | 0.3.0.1 | 2.14 <br> 2.15 <br> 2.16 <br> 2.17 |
```
```suggestion
| main | 9.4.x <br> 9.5.x <br> 9.7.x <br> 9.8.x | 3.9 <br> 3.10 <br> 3.11 <br> 3.12 | 0.3.0.1 | 2.14 <br> 2.15 <br> 2.16 <br> 2.17 |
```

**Review by meenakshidembi691 — APPROVED** [review]

**Review by trisha-dell — DISMISSED** [review]

**Review by meenakshidembi691 — APPROVED** [review]

**Review by Bhavneet-Sharma — APPROVED** [review]

**Review by meenakshidembi691 — APPROVED** [review]

**Pull request merged**